lib/index.js

```diff
@@ -84,9 +85,17 @@ VM.prototype.copy = function () {
  * Loads precompiled contracts into the state
  */
 VM.prototype.loadCompiled = function (address, src, cb) {
-  this.trie.db.put(address, src, cb)
+  var self = this
+  return new Promise(function (resolve, reject) {
+    cb = cb || function (err, result) { err ? reject(err) : resolve(result) }
```

Comment:
I don't think it is a good idea to return a promise which will never resolve - that is the case when a callback is passed (e.g. the old-world compatibility mode).

Wouldn't it make sense keeping the old interface unchanged and adding a promise API in front (or the other way around).

Also I'd be supportive internally moving to promises, which is a lot of work, and that would mean that the call back interface would need become the wrapper API.

Reply:
Can't we just put the switch in an if statement to have a more clean implementation with not returning an unresolved promise and at the same time keep the original idea and not have an additional/second API?

```javascript
VM.prototype._populateCache = function (addresses, cb) {
  self.stateManager.warmCache(addresses, cb)
}

VM.prototype.populateCache = function (addresses, cb) {
  if (!cb) {
    var self = this
    return new Promise(function (resolve, reject) {
      cb = function (err, result) { err ? reject(err) : resolve(result) }
      self._populateCache(addresses, cb)
    })
  } else {
    self._populateCache(addresses, cb)
  }
}
```